apex/functions/sign-xpi/README.rst

```diff
@@ -0,0 +1,58 @@
+sign_xpi
+========
+
+This function is responsible for "signing" an XPI for a system or privileged addon. This means taking a plain, unsigned
+system or privileged addon, and adding a signature to it. The signature is provided by `Autograph
+<https://github.com/mozilla-services/autograph/>`_, the Mozilla keyring service.
+
+Input
+=====
+
+This lambda expects a context which should provide this information:
+
+.. code-block:: json
+
+    {
+        "autograph": {
+            "serverUrl": "http://localhost:8888/",  // or whatever
+            "hawkId": "hawk-id-for-autograph",
+            "hawkSecret": "hawk-secret-for-autograph",
+            "keyId": "autograph-signer-key-id"
+        },
+        "outputBucket": "some-s3-bucket"
+    }
+
+The lambda is designed to sign only one category of addons: either system addons, or privileged extensions. To sign
+both, deploy the lambda twice with two sets of Autograph credentials.
+
+A "sign event" is taken as input, which should conform to this format:
+
+.. code-block:: json
+
+    {
+        "source": {
+            "bucket": "aws-s3-bucket",
+            "key": "some-xpi-filename.xpi"
+        },
+        "checksum": "sha256 of the object specified by source"
+    }
+
+As an alternative to passing an S3 bucket/key in the source, you can also pass a ``"url"`` field, which will be fetched by the lambda.
+
+Output
+======
+
+The lambda will sign the XPI and upload it to the S3 bucket specified in its context. (N.B. that you cannot change this bucket, because signing any XPI whatsoever is safe only if access to signed XPIs is trusted.) The return value of this lambda will be an object like:
+
+.. code-block:: json
+
+    {
+        "bucket": "s3-output-bucket",
```

Comment:
The design is for the two buckets to be different, and especially to have different permissions: the input bucket can be any bucket as long as the lambda can read from it; only the lambda can write to the output bucket, and very few authorized people should be able to read from it.